gdx/src/com/badlogic/gdx/graphics/g3d/loader/ObjLoader.java

```diff
@@ -439,4 +420,58 @@ public ModelMaterial getMaterial (final String name) {
 		materials.add(mat);
 		return mat;
 	}
-}
+
+	private static class CurrentMaterial {
+		String materialName = "default";
+		Color ambientColor = null;
+		Color diffuseColor = Color.WHITE;
+		Color specularColor = Color.WHITE;
+		float opacity = 1.f;
+		float shininess = 0.f;
+		String alphaTexFilename = null;
+		String ambientTexFilename = null;
+		String diffuseTexFilename = null;
+		String shininessTexFilename = null;
+		String specularTexFilename = null;
+
+		public ModelMaterial build () {
+			ModelMaterial mat = new ModelMaterial();
+			mat.id = materialName;
+			mat.ambient = ambientColor == null ? null : new Color(ambientColor);
+			mat.diffuse = new Color(diffuseColor);
+			mat.specular = new Color(specularColor);
+			mat.opacity = opacity;
+			mat.shininess = shininess;
+			addTexture(mat, alphaTexFilename, ModelTexture.USAGE_TRANSPARENCY);
+			addTexture(mat, ambientTexFilename, ModelTexture.USAGE_AMBIENT);
+			addTexture(mat, diffuseTexFilename, ModelTexture.USAGE_DIFFUSE);
+			addTexture(mat, specularTexFilename, ModelTexture.USAGE_SPECULAR);
+			addTexture(mat, shininessTexFilename, ModelTexture.USAGE_SHININESS);
+
+			return mat;
+		}
+
+		private void addTexture (ModelMaterial mat, String texFilename, int usage) {
+			if (texFilename == null) {
```

Comment:
if texFilename != null then add to material would be better in term of readability